### DISCUSSION TITLE
Bulb turns red after MQTT reconnect

### DISCUSSION THREAD

**Discussion by lurendrejer** [arendst/Tasmota#15993] [Q&A] [answered]
Hi.

My shelly bulbs turns red after reconnecting to the MQTT broker, has anyone had anything like this?
I startet noticing that every once in a while a single bulb would have turned RED over night.
At first i thought it was some sort of WIFI-status LED functionality, but that doesn't seem to be the case.

If I block the bulbs MQTT connection (or misconfigure it and configure it again) the bulb turns red, after subscribing.
I have tried sending another color as the last message to the bulb to se if it was some sort of retainment - but it consistently turns back to RED when reconnecting.
The broker is an AEDES MQTT running in node-red.

![image](https://user-images.githubusercontent.com/30922656/179346991-fc24313b-8878-44ba-a4e8-6a2361e2f768.png)

**Comment by lurendrejer** [comment, marked as answer]
LedPower 0 seems to solve the 'issue'.
I guess RED is: MQTT problem and BLUE is WIFI-problem for bulbs.

**Reply by sfromis** [reply]
To see what happens, check the log after setting `Weblog 3` for details. I suspect some automation or retained messages acting after connection.

**Reply by lurendrejer** [reply]
Oh, thank you - will do :)

**Reply by lurendrejer** [reply]
Today a single bulb was blue - when i woke up.

I don't have any automations that would set the bulb to blue (or RED for that matter)

I'm still looking into this - thank you for the weblog-tip @sfromis

**Reply by lurendrejer** [reply]
Well, it does seem like something is retained - why though?
Set weblog to 3 - and kicked the bulb from the wifi-network

09:12:04.711 CMD: weblog 3
09:12:04.717 MQT: stat/bulbname2/RESULT = {"WebLog":3}
09:12:05.034 CFG: Saved to flash at FA, Count 666, Bytes 4096
09:12:14.732 SRC: MQTT
09:12:14.735 CMD: Grp 0, Cmd 'WHITE', Idx 1, Len 3, Pld 100, Data '100'
09:12:14.743 MQT: stat/bulbname2/RESULT = {"POWER":"ON","Dimmer":100,"Color":"000000FF","HSBColor":"173,100,0","White":100,"Channel":[0,0,0,100]}
09:12:15.043 CFG: Saved to flash at F9, Count 667, Bytes 4096
09:12:15.377 SRC: MQTT
09:12:15.379 CMD: Grp 0, Cmd 'DIMMER', Idx 1, Len 1, Pld 0, Data '0'
09:12:15.383 SRC: Light
09:12:15.390 MQT: stat/bulbname2/RESULT = {"POWER":"OFF","Dimmer":0,"Color":"00000000","HSBColor":"173,100,0","White":0,"Channel":[0,0,0,0]}
09:12:16.062 CFG: Saved to flash at F8, Count 668, Bytes 4096
09:12:28.521 MQT: Attempting connection...
09:12:28.524 DNS: Unable to resolve 'mqtt.#####.lan'
09:12:28.526 MQT: Connect failed to mqtt.#####.lan:1883, rc -5. Retry in 10 sec
09:12:41.541 MQT: Attempting connection...
09:12:41.579 MQT: Connected
09:12:41.584 MQT: tele/bulbname2/LWT = Online (retained)
09:12:41.587 MQT: cmnd/bulbname2/POWER = 
09:12:41.590 MQT: Subscribe to cmnd/bulbname2/#
09:12:41.592 MQT: Subscribe to cmnd/tasmotas/#
09:12:41.595 MQT: Subscribe to cmnd/bulbname2_fb/#
09:12:41.950 SRC: MQTT
09:12:41.953 CMD: Grp 0, Cmd 'WHITE', Idx 1, Len 5, Pld -99, Data ' 100'
09:12:41.959 MQT: stat/bulbname2/RESULT = {"White":0}
09:12:41.000 SRC: MQTT
09:12:41.002 CMD: Grp 1, Cmd 'LEDPOWER', Idx 1, Len 3, Pld -99, Data ' 0'
09:12:42.002 MQT: stat/bulbname2/RESULT = {"LedPower1":"OFF"}
09:12:42.044 SRC: MQTT
09:12:42.047 CMD: Grp 0, Cmd 'DIMMER', Idx 1, Len 3, Pld -99, Data ' 0'
09:12:42.051 MQT: stat/bulbname2/RESULT = {"Dimmer":0}
09:12:42.094 SRC: MQTT
09:12:42.097 CMD: Grp 1, Cmd 'SETOPTION', Idx 31, Len 3, Pld -99, Data ' 1'
09:12:42.101 MQT: stat/bulbname2/RESULT = {"SetOption31":"ON"}
09:12:42.144 SRC: MQTT
09:12:42.147 CMD: Grp 0, Cmd 'COLOR', Idx 1, Len 10, Pld -99, Data ' FFDE3E00'
09:12:42.151 SRC: Light
09:12:42.163 MQT: stat/bulbname2/RESULT = {"POWER":"ON","Dimmer":100,"Color":"00FFDE3E","HSBColor":"173,100,100","White":24,"Channel":[0,100,87,24]}
09:12:43.071 CFG: Saved to flash at F7, Count 669, Bytes 4096
09:12:58.647 CMD: weblog 2
09:12:58.657 MQT: stat/bulbname2/RESULT = {"WebLog":2}
09:14:30.429 MQT: stat/bulbname2/RESULT = {"POWER":"ON","Dimmer":100,"Color":"FFDE3E00","HSBColor":"50,76,100","White":0,"Channel":[100,87,24,0]}

**Reply by sfromis** [reply]
The answer to "why" is "because you did it". Maybe through some tool, but not something controlled by Tasmota. If you have experimented with retained commands, the last instance of those will still be retained even though you may have stopped publishing with retain. Generally, old retained messages can be cleared by publishing an empty retained message to the same topic, or through some MQTT tools.

**Reply by lurendrejer** [reply]
Pretty sure about it being my own fault :)

I use the tasmotas -group topic to control all lights and set some parameters.
I think these messages might be retained - since 'SetOption31' for instance is in the weblog3-messages.

**Reply by lurendrejer** [reply]
You were right on, retained messages is the problem.
I might have sent a retained message from MQTT-explorer to the group topic at some time :)

![image](https://user-images.githubusercontent.com/30922656/179480712-c019e08b-4abb-41f0-a45b-b31eaf39df1a.png)

**Reply by lurendrejer** [reply]
And thank you for the help, it is much appriciated. :)

**Reply by sfromis** [reply]
Better abstain from retained commands, unless you know *exactly* why you want it, and have prepared to deal with the many pitfalls. For most use cases, you should publish all commands to the same topic with retain after doing it once, or clear the stored retained message when you stop publishing with retain.

**Reply by lurendrejer** [reply]
Must have been a late-night brain-fart (while integrating with homekit), and thank you again.
I'm not really sure how i mark your reply as an answer.
